webapp/src/features/adminOrganizationTabUser/adminOrganizationTabUserSlice.js

```diff
@@ -0,0 +1,245 @@
+);
+
+export const userDeleteSingle = createAsyncThunk(
+  "adminOrganizationTabUser/userDeleteSingle",
+  async (payload, { getState, dispatch }) => {
+    const { user, orgPatchJson, selectedOrg, fetchPatchOrganization, updateTableData } = payload;
+    const currentState = getState();
+    const token = selectToken(currentState);
+
+    let promises = [];
+    const userData = (await getUserData(user.email, token)).body;
+    if (userData?.user_data?.organizations?.length > 1) {
+      const userRole = { email: user.email, role: user.role };
+      let patchJson = { ...orgPatchJson };
+
+      patchJson.users_to_remove = [userRole];
+      promises.push(fetchPatchOrganization(patchJson));
+    } else {
+      alert("Cannot remove User " + user.email + " from " + selectedOrg + " because they must belong to at least one organization.");
+    }
+    Promise.all(promises).then(() => {
+      dispatch(refresh({ selectedOrg, updateTableData }));
+    });
+  },
+  { condition: (_, { getState }) => selectToken(getState()) }
+);
+
+export const userDeleteMultiple = createAsyncThunk(
+  "adminOrganizationTabUser/userDeleteMultiple",
+  async (payload, { getState, dispatch }) => {
+    const { users, orgPatchJson, selectedOrg, fetchPatchOrganization, updateTableData } = payload;
+    const currentState = getState();
+    const token = selectToken(currentState);
+
+    let promises = [];
+    for (const user of users) {
+      const userData = (await getUserData(user.email, token)).body;
+      if (userData?.user_data?.organizations?.length > 1) {
+        const userRole = { email: user.email, role: user.role };
+        let patchJson = { ...orgPatchJson };
+        patchJson.rsus_to_remove = [userRole];
+        promises.push(fetchPatchOrganization(patchJson));
+      } else {
+        alert("Cannot remove User " + user.email + " from " + selectedOrg + " because they must belong to at least one organization.");
+      }
+    }
+    Promise.all(promises).then(() => {
+      dispatch(refresh({ selectedOrg, updateTableData }));
+    });
+  },
+  { condition: (_, { getState }) => selectToken(getState()) }
+);
+
+export const userAddMultiple = createAsyncThunk(
+  "adminOrganizationTabUser/userAddMultiple",
+  async (payload, { dispatch }) => {
+    const { userList, orgPatchJson, selectedOrg, fetchPatchOrganization, updateTableData } = payload;
+    let promises = [];
+    for (const user of userList) {
+      const patchJson = { ...orgPatchJson };
+      const userRole = { email: user?.email, role: user?.role };
+      patchJson.users_to_add = [userRole];
+      promises.push(fetchPatchOrganization(patchJson));
+    }
+    Promise.all(promises).then(() => {
+      dispatch(refresh({ selectedOrg, updateTableData }));
+    });
+  },
+  { condition: (payload, { getState }) => selectToken(getState()) && payload.userList != [] }
+);
+
+export const userbulkEdit = createAsyncThunk(
+  "adminOrganizationTabUser/userAddMultiple",
```

Comment:
Is this ok to have the same value as userAddMultiple thunk above? Are both functions needed?

Reply:
re-named to match thunk name